apps/payments/next/app/[locale]/layout.tsx

```diff
@@ -34,7 +37,16 @@ export default async function RootProviderLayout({
       fetchedMessages={fetchedMessages}
       nonce={nonce}
     >
-      {children}
+      <Header
+        auth={{
+          user: session?.user,
+          signOut: async () => {
+            'use server';
+            await signOut();
+          },
```

Comment:
The sign out function needs to either clean up the existing cart the user is currently on, or create a new cart (easiest is to redirect the customer to `/new`).

The cart is used to keep the users state for their checkout session. If the user signs out and the cart is left as is, then the cart state and ui state are out of sync. Specifically thinking of `cart.uid` as the most important, but could be other fields as well such as `stripeCustomerId`, `stripeSubscriptionId`, etc.

Because of this, I favor a new cart being created after signout, and letting the user start fresh.

Reply:
Redirecting here got pretty weird, I think there's a fair bit being obscured by the nextjs server-client boundary here, and some unacknowledged race conditions happening. I've gotten it working by passing the client route props through to the `<Header>` component's `signOut` callback, and constructing the redirect route there.

Reply:
(redirecting to the `/new` route causes some additional weirdness where we trigger the new cart creation but dont navigate to it. `/landing` works just fine)